Acceptance-Tests.md

```diff
@@ -55,34 +67,34 @@ Docker:
 > Note: Naming convention for directories / files is kebab case, where each word is separated by a (-)
 
-2) Within the user directory, create a new file for each test. For ex. `create-new-topic.spec.js` and `delete-topic.spec.js` for `Topic Manager` user. And these top-level test contains single user stories checking their test steps and expectations mentioned in the [testing spreadsheet](https://docs.google.com/spreadsheets/d/1O8EHiSAGrG0yoNUBz9E4DIwKNS8Rfsv_ffC4k1WK5jc/edit?usp=sharing).
+2) Within the user directory, create a new file for each test. For ex. `create-new-topic.spec.ts` and `delete-topic.spec.ts` for `Topic Manager` user. And these top-level test contains single user stories checking their test steps and expectations mentioned in the [testing spreadsheet](https://docs.google.com/spreadsheets/d/1O8EHiSAGrG0yoNUBz9E4DIwKNS8Rfsv_ffC4k1WK5jc/edit?usp=sharing).
 
-3) The functionality of the top-level tests for each user-type is defined in the `user-utilities` directory. For ex. the blog editor tests are written within the `spec/blog-editor-tests` directory, and the functionality of the tests are defined in the `user-utilities/blog-post-editor-utils.js` file.
-> Note: A utility file is maintained for each user type. The purpose of maintaining this file is to add methods specific to that user on top of the already provided basic methods. This file maintains a user class which is extended from the base class of puppeteer-utils.js . For ex. blog-post-admin-utils.js have a class e2eBlogPostAdmin which have methods like `createDraftBlogPostWithTitle`, `deleteDraftBlogPostWithTitle` etc. specific to Blog Admin only.
+3) The functionality of the top-level tests for each user-type is defined in the `user-utilities` directory. For ex. the blog editor tests are written within the `spec/blog-editor-tests` directory, and the functionality of the tests are defined in the `user-utilities/blog-post-editor-utils.ts` file.
+> Note: A utility file is maintained for each user type. The purpose of maintaining this file is to add methods specific to that user on top of the already provided basic methods. This file maintains a user class which is extended from the base class of puppeteer-utils.ts . For ex. blog-post-admin-utils.ts have a class BlogPostEditor which have methods like `createDraftBlogPostWithTitle`, `deleteDraftBlogPostWithTitle` etc. specific to Blog Admin only.
 
 
-4) The utility files are imported in the top-level test files and the methods are called to perform the required actions. For ex. in the `create-draft-and-delete-draft-blog-post.spec.js` file, the `createDraftBlogPostWithTitle` method is called to create a draft blog post with a given title. The `deleteDraftBlogPostWithTitle` method is called to delete the draft blog post with the given title.
```

Comment:
Should also reference the user to create a UserFactory inside each utils file which the purpose is to instantiate a new class of that type (e.g. export let QuestionAdminFactory = (): QuestionAdmin => new QuestionAdmin();)

Reply:
(4) The utility files are imported into the top-level test files, and the methods are called to perform the required actions. For example, in the `try-to-publish-a-duplicate-blog-post-and-get-blocked.spec.ts` file, the `createNewBlogPostWithTitle` method is called to create a new blog post with the given title. Additionally, the `expectUserUnableToPublishBlogPost` method is called to check if the user is unable to publish a blog post. To facilitate instantiation of classes, each utils file also includes a `UserFactory` function. This function's purpose is to instantiate a new class of the corresponding type. For instance, `export let QuestionAdminFactory = (): QuestionAdmin => new QuestionAdmin();` would create a `QuestionAdmin` instance.

@jnvtnguyen PTAL.

Reply:
each utils file also -> each utils file should also include a UserFactory function.

Reply:
done